Fix plugin unit test for MacOS

When running `mvn clean install` in `its/plugin/`, I get the following error on **MacOS** for the test `TypeScriptSensorTest.java#should_not_send_content`:

```
[ERROR] Tests run: 20, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.348 s <<< FAILURE! - in org.sonar.plugins.javascript.eslint.TypeScriptSensorTest
[ERROR] should_not_send_content  Time elapsed: 0.046 s  <<< FAILURE!
org.mockito.exceptions.verification.WantedButNotInvoked:

Wanted but not invoked:
eslintBridgeServerMock.analyzeWithProgram(
    <Capturing argument>
);
-> at org.sonar.plugins.javascript.eslint.TypeScriptSensorTest.should_not_send_content(TypeScriptSensorTest.java:303)

However, there were exactly 4 interactions with this mock:
eslintBridgeServerMock.startServerLazily(
    org.sonar.api.batch.sensor.internal.SensorContextTester@54224125
);
-> at org.sonar.plugins.javascript.eslint.AbstractEslintSensor.execute(AbstractEslintSensor.java:66)

eslintBridgeServerMock.initLinter(
    [no-all-duplicated-branches],
    [],
    []
);
-> at org.sonar.plugins.javascript.eslint.TypeScriptSensor.analyzeFiles(TypeScriptSensor.java:88)

eslintBridgeServerMock.createProgram(
    org.sonar.plugins.javascript.eslint.EslintBridgeServer$TsProgramRequest@7c013560
);
-> at org.sonar.plugins.javascript.eslint.AnalysisWithProgram.analyzeFiles(AnalysisWithProgram.java:87)

eslintBridgeServerMock.deleteProgram(
    TsProgram{programId='1', files=[/var/folders/bj/b0jbqvls1kld4kqgy_07rcnc0000gp/T/junit12362648857168675886/dir/file.ts], projectReferences=[]}
);
-> at org.sonar.plugins.javascript.eslint.AnalysisWithProgram.analyzeFiles(AnalysisWithProgram.java:97)


	at org.sonar.plugins.javascript.eslint.TypeScriptSensorTest.should_not_send_content(TypeScriptSensorTest.java:303)
```

This issue was introduced by https://github.com/SonarSource/SonarJS/pull/3165.
A proposed fix was done in https://github.com/SonarSource/SonarJS/pull/3173, but as it may have side effects, we opened this issue to work on it instead.

--- a/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TypeScriptSensorTest.java
+++ b/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TypeScriptSensorTest.java
@@ -230,11 +230,14 @@ class TypeScriptSensorTest {
     verify(eslintBridgeServerMock, never()).analyzeTypeScript(any());
     verify(eslintBridgeServerMock, never()).analyzeWithProgram(any());
 
+    String errorLog;
     if (isWindows()) {
-      assertThat(logTester.logs(LoggerLevel.ERROR)).contains("Provided tsconfig.json path doesn't exist. Path: '" + baseDir.toRealPath().resolve("wrong.json") + "'"); // toRealPath avoids 8.3 paths on Windows
+      // toRealPath avoids 8.3 paths on Windows
+      errorLog = "Provided tsconfig.json path doesn't exist. Path: '" + baseDir.toRealPath().resolve("wrong.json") + "'";
     } else {
-      assertThat(logTester.logs(LoggerLevel.ERROR)).contains("Provided tsconfig.json path doesn't exist. Path: '" + baseDir.resolve("wrong.json") + "'"); // toRealPath avoids 8.3 paths on Windows
+      errorLog = "Provided tsconfig.json path doesn't exist. Path: '" + baseDir.resolve("wrong.json") + "'";
     }
+    assertThat(logTester.logs(LoggerLevel.ERROR)).contains(errorLog);
     
   }
 

--- a/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TypeScriptSensorTest.java
+++ b/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TypeScriptSensorTest.java
@@ -230,12 +230,12 @@ class TypeScriptSensorTest {
     verify(eslintBridgeServerMock, never()).analyzeTypeScript(any());
     verify(eslintBridgeServerMock, never()).analyzeWithProgram(any());
 
-    String errorLog;
+    String errorLog = "Provided tsconfig.json path doesn't exist. Path: '%s'";
     if (isWindows()) {
       // toRealPath avoids 8.3 paths on Windows
-      errorLog = "Provided tsconfig.json path doesn't exist. Path: '" + baseDir.toRealPath().resolve("wrong.json") + "'";
+      errorLog = String.format(errorLog, baseDir.toRealPath().resolve("wrong.json"));
     } else {
-      errorLog = "Provided tsconfig.json path doesn't exist. Path: '" + baseDir.resolve("wrong.json") + "'";
+      errorLog = String.format(errorLog, baseDir.resolve("wrong.json"));
     }
     assertThat(logTester.logs(LoggerLevel.ERROR)).contains(errorLog);
     

--- a/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TypeScriptSensorTest.java
+++ b/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TypeScriptSensorTest.java
@@ -230,13 +230,24 @@ class TypeScriptSensorTest {
     verify(eslintBridgeServerMock, never()).analyzeTypeScript(any());
     verify(eslintBridgeServerMock, never()).analyzeWithProgram(any());
 
-    assertThat(logTester.logs(LoggerLevel.ERROR)).contains("Provided tsconfig.json path doesn't exist. Path: '" + baseDir.toRealPath().resolve("wrong.json") + "'"); // toRealPath avoids 8.3 paths on Windows
+    if (isWindows()) {
+      assertThat(logTester.logs(LoggerLevel.ERROR)).contains("Provided tsconfig.json path doesn't exist. Path: '" + baseDir.toRealPath().resolve("wrong.json") + "'"); // toRealPath avoids 8.3 paths on Windows
+    } else {
+      assertThat(logTester.logs(LoggerLevel.ERROR)).contains("Provided tsconfig.json path doesn't exist. Path: '" + baseDir.resolve("wrong.json") + "'"); // toRealPath avoids 8.3 paths on Windows
+    }
+    
   }
 
   private SensorContextTester createSensorContext(Path baseDir) throws IOException {
-    // toRealPath avoids 8.3 paths on Windows, which clashes with tests where test file location is checked
-    // https://en.wikipedia.org/wiki/8.3_filename
-    SensorContextTester ctx = SensorContextTester.create(baseDir.toRealPath());
+    SensorContextTester ctx = null;
+    if (isWindows()) {
+      // toRealPath avoids 8.3 paths on Windows, which clashes with tests where test file location is checked
+      // https://en.wikipedia.org/wiki/8.3_filename
+      ctx = SensorContextTester.create(baseDir.toRealPath());
+    } else {
+      ctx = SensorContextTester.create(baseDir);
+    }
+    
     ctx.fileSystem().setWorkDir(workDir);
     return ctx;
   }
@@ -636,6 +647,11 @@ class TypeScriptSensorTest {
 
   private void setSonarLintRuntime(SensorContextTester context) {
     context.setRuntime(SonarRuntimeImpl.forSonarLint(Version.create(8, 9)));
+  }
+
+  private boolean isWindows() {
+    var osName = System.getProperty("os.name");
+    return osName.toLowerCase().startsWith("win");
   }
 
   private static TypeScriptChecks checks(String... ruleKeys) {

